From f71051e8f1add0b893ffaa9a799625017978e7f8 Mon Sep 17 00:00:00 2001
From: Frank Dana <ferdnyc@gmail.com>
Date: Thu, 20 Aug 2020 21:13:42 -0400
Subject: [PATCH] Tests: Don't enable HW accel as side-effect (#557)

The Settings test attempts to test the class by changing settings
variables and reading them back again. Problem is, that affects
the REST of the unit tests. So instead of enabling HW accel and
causing crashes, we'll diddle something innocuous, like OMP_THREADS.
---
 tests/Settings_Tests.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/Settings_Tests.cpp b/tests/Settings_Tests.cpp
index b63b56a8..65bd66a6 100644
--- a/tests/Settings_Tests.cpp
+++ b/tests/Settings_Tests.cpp
@@ -41,7 +41,7 @@ TEST(Settings_Default_Constructor)
 	// Create an empty color
 	Settings *s = Settings::Instance();
 
-	CHECK_EQUAL(0, s->HARDWARE_DECODER);
+	CHECK_EQUAL(12, s->OMP_THREADS);
 	CHECK_EQUAL(false, s->HIGH_QUALITY_SCALING);
 	CHECK_EQUAL(false, s->WAIT_FOR_VIDEO_PROCESSING_TASK);
 }
@@ -50,15 +50,15 @@ TEST(Settings_Change_Settings)
 {
 	// Create an empty color
 	Settings *s = Settings::Instance();
-	s->HARDWARE_DECODER = 1;
+	s->OMP_THREADS = 8;
 	s->HIGH_QUALITY_SCALING = true;
 	s->WAIT_FOR_VIDEO_PROCESSING_TASK = true;
 
-	CHECK_EQUAL(1, s->HARDWARE_DECODER);
+	CHECK_EQUAL(8, s->OMP_THREADS);
 	CHECK_EQUAL(true, s->HIGH_QUALITY_SCALING);
 	CHECK_EQUAL(true, s->WAIT_FOR_VIDEO_PROCESSING_TASK);
 
-	CHECK_EQUAL(1, s->HARDWARE_DECODER);
+	CHECK_EQUAL(8, Settings::Instance()->OMP_THREADS);
 	CHECK_EQUAL(true, Settings::Instance()->HIGH_QUALITY_SCALING);
 	CHECK_EQUAL(true, Settings::Instance()->WAIT_FOR_VIDEO_PROCESSING_TASK);
 }
